libc/string_lenchr.c

```diff
@@ -150,9 +169,259 @@ TEST(string_len, big)
+}
+
+
+TEST(string_chr, ascii_cases)
+{
+
+	char *torn = "foo\0bar",
+		 *whites = " \v\t\r\n",
+		 *xwhites = " \v\t\r\nx";
+
+	int sz, i;
+	char str[INT8_MAX] = { 0 };
+	const char *format = "\399\45\43\0";
+
+	for (i = 1; i < sizeof(str); i++) {
+		str[i - 1] = i;
+	}
+	str[i - 1] = '\0';
+
+	sz = sizeof(str) - 1;
+
+	TEST_ASSERT_EQUAL_STRING(str, strchr(str, str[0]));
+	TEST_ASSERT_EQUAL_STRING(str, strrchr(str, str[0]));
+	TEST_ASSERT_EQUAL_STRING(str, memchr(str, str[0], sz + 1));
+
+	TEST_ASSERT_EQUAL_STRING(&str[sz / 2], strchr(str, str[sz / 2]));
+	TEST_ASSERT_EQUAL_STRING(&str[sz / 2], strrchr(str, str[sz / 2]));
+	TEST_ASSERT_EQUAL_STRING(&str[sz / 2], memchr(str, str[sz / 2], sz + 1));
+
+	TEST_ASSERT_EQUAL_STRING(&str[sz], strchr(str, str[sz]));
+	TEST_ASSERT_EQUAL_STRING(&str[sz], strrchr(str, str[sz]));
+	TEST_ASSERT_EQUAL_STRING(&str[sz], memchr(str, str[sz], sz + 1));
+
+	/* Checking correct working of null terminating point */
+	sz = strlen(torn) + 1;
+	TEST_ASSERT_EQUAL_STRING("", strchr(torn, '\0'));
+	TEST_ASSERT_EQUAL_STRING("", strrchr(torn, '\0'));
+	TEST_ASSERT_EQUAL_STRING("", memchr(torn, '\0', sz));
+
+	TEST_ASSERT_EQUAL_STRING("foo", strchr(torn, 'f'));
+	TEST_ASSERT_EQUAL_STRING("foo", strrchr(torn, 'f'));
+	TEST_ASSERT_EQUAL_STRING("foo", memchr(torn, 'f', sz));
+
+	TEST_ASSERT_EQUAL_STRING(NULL, strchr(torn, 'b'));
+	TEST_ASSERT_EQUAL_STRING(NULL, strrchr(torn, 'b'));
+	TEST_ASSERT_EQUAL_STRING(NULL, memchr(torn, 'b', sz));
+
+	/* 99%# decode ASCII after number 9 which is not in ASCII chain in this case */
+	TEST_ASSERT_EQUAL_STRING("99%#", strchr(format, '9'));
+	TEST_ASSERT_EQUAL_STRING("9%#", strrchr(format, '9'));
+	TEST_ASSERT_EQUAL_STRING("99%#", memchr(format, '9', 7));
+
+	/* Testing if we can get a number from ASCII if it represents the symbol of him */
+	TEST_ASSERT_EQUAL_STRING(NULL, strchr(format, '4'));
+	TEST_ASSERT_EQUAL_STRING(NULL, strrchr(format, '4'));
+	TEST_ASSERT_EQUAL_STRING(NULL, memchr(format, '4', 7));
+
+	/* Checking if we can get a white sign from a string full of different white signs */
+	sz = ptrContentSize(whites);
+	TEST_ASSERT_EQUAL_STRING("\n", strchr(whites, '\n'));
+	TEST_ASSERT_EQUAL_STRING("\n", strrchr(whites, '\n'));
+	TEST_ASSERT_EQUAL_STRING("\n", memchr(whites, '\n', sz));
+
+	/* Checking possibility and correction of value after white signs */
+	sz = ptrContentSize(xwhites);
+	TEST_ASSERT_EQUAL_STRING("x", strchr(xwhites, 'x'));
+	TEST_ASSERT_EQUAL_STRING("x", strrchr(xwhites, 'x'));
+	TEST_ASSERT_EQUAL_STRING("x", memchr(xwhites, 'x', sz));
+
+	TEST_ASSERT_EQUAL_STRING(whites, strchr(whites, ' '));
+	TEST_ASSERT_EQUAL_STRING(whites, strrchr(whites, ' '));
+	TEST_ASSERT_EQUAL_STRING(whites, memchr(whites, ' ', strlen(whites)));
+
+	TEST_ASSERT_EQUAL_STRING(NULL, strchr(whites, 'x'));
+	TEST_ASSERT_EQUAL_STRING(NULL, strrchr(whites, 'x'));
+	TEST_ASSERT_EQUAL_STRING(NULL, memchr(whites, 'x', strlen(whites)));
```

Comment:
We decided to split test cases to smaller ones in `chr` tests, so please create a separate test case `whitespacees` and place the mentioned code there. Above that I know that I suggested you the solution with `xwhites`, but now I see that checking \0 as the last character would be sufficient, so please take the `" \v\t\r\n"` string and est chr functions for space, termination char, some char in the middle, for example \t and x which is not present in the mentioned string.

```suggestion
```